### PR TITLE
[MM-20164] Member to Admin not updating has been fixed in Channnel Member Dropdown

### PR DESCRIPTION
**Pull request by hahmadia**
#### Summary

If you are a regular member looking at the `view members` section; if you are promoted to a system admin, you will not see the changes reflect immediately. This PR fixes this problem. 

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-20164

### Pre-Fix
MM-20164-PreFix

### Post-Fix
MM-20164-PostFix

### CODE REVIEW

**Comment by hahmadia** [comment]
My previous fix to the problem wasn't the best way. I decided to update the new roles in the websocket instead. I also have added cypress test.

File: e2e/cypress/integration/channel/user_to_admin_updates_manage_channel_members_modal_spec.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+import users from '../../fixtures/users.json';
+
+const demoteToMember = (user) => {
+    const baseUrl = Cypress.config('baseUrl');
+    cy.externalRequest({user: users.sysadmin, method: 'put', baseUrl, path: `users/${user.id}/roles`, data: {roles: 'system_user'}});
+};
+
+const promoteToSysAdmin = (user) => {
+    const baseUrl = Cypress.config('baseUrl');
```
Comment (saturninoabril) [inline e2e/cypress/integration/channel/user_to_admin_updates_manage_channel_members_modal_spec.js]: Just like to chime in but baseUrl is actually no use here. It's already in the command itself - https://github.com/mattermost/mattermost-webapp/blob/master/e2e/cypress/support/task_commands.js#L41

Comment (hahmadia) [reply e2e/cypress/integration/channel/user_to_admin_updates_manage_channel_members_modal_spec.js]: Removed. Thanks

File: e2e/cypress/integration/channel/user_to_admin_updates_manage_channel_members_modal_spec.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+describe('View Members modal', () => {
+    it('MM-20164 - Going from a Member to an Admin should update the modal', () => {
+        cy.apiLogin('user-1');
+        cy.apiGetMe().then((res) => {
+            // # Make user a regular member
+            demoteToMember(res.body);
+
+            // # Visit Town square and go to view members modal
+            cy.visit('/');
+            cy.get('#sidebarItem_town-square').click({force: true});
```
Comment (saturninoabril) [inline e2e/cypress/integration/channel/user_to_admin_updates_manage_channel_members_modal_spec.js]: Should this be tested on town-square?

File: e2e/cypress/integration/channel/user_to_admin_updates_manage_channel_members_modal_spec.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+            cy.get('#member_popover').click();
+            cy.findByTestId('membersModal').click();
+
+            // * Check to see if no drop down menu exists
+            cy.findAllByTestId('userListItemActions').then((el) => {
```
Comment (saturninoabril) [inline e2e/cypress/integration/channel/user_to_admin_updates_manage_channel_members_modal_spec.js]: Looks like this is not correct. `userListItemActions` can be found always.  Should probably check for its content to effectively test it.  Same case when promoting to admin.

Comment (saturninoabril) [reply e2e/cypress/integration/channel/user_to_admin_updates_manage_channel_members_modal_spec.js]: Agreed offline to just check text content when user is promoted.

Comment (hahmadia) [reply e2e/cypress/integration/channel/user_to_admin_updates_manage_channel_members_modal_spec.js]: Done!

**Pull request merged**